src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,153 @@
 package core.basesyntax;
+//import jdk.internal.util.ArraysSupport;
+
+import java.util.Arrays;
+import java.util.NoSuchElementException;
 
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final Object[] EMPTY_ARRAYDATA = {};
+    private T[] arrayData = (T[]) new Object[DEFAULT_CAPACITY];
+    private int size;
+    private int cursor;
+
+    private String outOfBoundsMsg(int index) {
+        return "Index: " + index + ", Size: " + size;
+    }
+
+    private Object[] grow(int minimalCapacity) {
+        int oldCapacity = arrayData.length;
+        double multiplicator = 1;
+        if (oldCapacity == size) {
+            multiplicator = 1.5;
+        }
+        if (oldCapacity > 0 || arrayData != EMPTY_ARRAYDATA) {
+            int newCapacity = (int) (oldCapacity * multiplicator + 1);
+            return arrayData = Arrays.copyOf(arrayData, newCapacity);
```

Comment:
![Screenshot 2022-06-24 at 14 53 37](https://user-images.githubusercontent.com/60258373/175529096-a0f39f96-1a0c-4c24-801c-4a3619c40828.png)
Before you create PR with your homework, you need to go through the checklist under the task, and correct all the points described there. The mentor will not check the work until the checklist items are corrected